cmd/soroban-cli/src/commands/contract/init.rs

```diff
@@ -637,8 +658,32 @@ mod tests {
     fn assert_contract_cargo_file_uses_workspace(project_dir: &Path, contract_name: &str) {
         let contract_dir = project_dir.join("contracts").join(contract_name);
         let cargo_toml_path = contract_dir.as_path().join("Cargo.toml");
-        let cargo_toml_str = read_to_string(cargo_toml_path).unwrap();
-        assert!(cargo_toml_str.contains("soroban-sdk = { workspace = true }"));
+        println!(
+            "Cargo.toml:\n{}",
+            read_to_string(cargo_toml_path.clone()).unwrap()
+        );
+        let toml = cargo_toml::Manifest::from_path(cargo_toml_path).unwrap();
+        if let Some(dep) = toml.dependencies.get("soroban-sdk").unwrap().detail() {
```

Comment:
One weird thing: I had to add this “if let“ here, because the hello-world Cargo.toml is evaluated as having no “dependencies.soroban-sdk”, which seems obviously wrong. I ran out of time to investigate yesterday, but it might be worth looking into more before we merge.

Reply:
Did the testing with `toml` crate instead of `cargo_toml`. Still using `cargo_toml` to construct the file, which works well, but for some reason when parsing the (correctly-formatted) files, it was parsing `dependencies.soroban-sdk.workspace = true` into its internal version of`dependencies.soroban-sdk.version = "20.3.2"`. Checking it with just `toml` demonstrates that it is well-formatted.